controllers/v1/observationsController.js

```diff
@@ -631,98 +631,180 @@ module.exports = class Observations extends Abstract {
-                    } else {
+                    let entitiesDocument = await sunbirdService.learnerLocationSearch( filterData );
 
+                    if ( !entitiesDocument.success || !entitiesDocument.data || !entitiesDocument.data.response.length > 0 ) {
+                        return resolve({
+                            "message" : messageConstants.apiResponses.ENTITY_NOT_FOUND,
+                            "result" : {
+                                "count":0,
+                                "data" : []
+                            }
+                        })
+                    }
+                    let entitiesData = entitiesDocument.data.response;
+                    
+                    if( entitiesData && entitiesData[0].type === result.entityType ) {
                         response.result = [];
-                        if( entitiesData[0] && entitiesData[0].entityType === result.entityType ) {
-
-                            if( entitiesData[0].metaInformation ) {
-                                
-                                if( entitiesData[0].metaInformation.name ) {
-                                    entitiesData[0]["name"] = entitiesData[0].metaInformation.name;
-                                }
-
-                                if( entitiesData[0].metaInformation.externalId ) {
-                                    entitiesData[0]["externalId"] = entitiesData[0].metaInformation.externalId;
-                                }
-
-                                if( entitiesData[0].metaInformation.addressLine1 ) {
-                                    entitiesData[0]["addressLine1"] = entitiesData[0].metaInformation.addressLine1;
-                                }
-
-                                if( entitiesData[0].metaInformation.addressLine2 ) {
-                                    entitiesData[0]["addressLine2"] = entitiesData[0].metaInformation.addressLine2;
-                                }
-
-                                if( entitiesData[0].metaInformation.districtName ) {
-                                    entitiesData[0]["districtName"] = entitiesData[0].metaInformation.districtName;
-                                }
-
-                                entitiesData[0] = _.pick(
-                                    entitiesData[0],
-                                    ["_id","name","externalId","addressLine1","addressLine2","districtName"]
-                                )
+                        
+                        let data = 
+                        await entitiesHelper.observationSearchEntitiesResponse(
+                            entitiesData,
+                            result.entities
+                        );
+
+                        response["message"] = messageConstants.apiResponses.ENTITY_FETCHED;
+
+                        response.result.push({
+                            "data" : data,
+                            "count" : 1
```

Comment:
@VISHNUDAS-tunerlabs why count is 1

Reply:
the data count will be always one here. old code also specified like this.